marshal/claim.go

```diff
@@ -278,6 +281,9 @@ func (c *claim) teardown(releasePartition bool) bool {
 		return false
 	}
 
+	// Kill the stopchan now which is a useful way of knowing we're quitting within selects
+	close(c.stopChan)
```

Comment:
Does it? My understanding of channels is that it's safe to be reading/closing them at the same time. You can't write to one you're closing, of course, but for a read-only channel used for signalling I thought it was safe?